tests/test_child_plugins.py

```diff
@@ -0,0 +1,236 @@
+from strax.testutils import *
+import pytest
+
+
+def test_child_plugin_config():
+    """
+    Test if options of parent plugin are correctly replaced by config
+    options of the child plugin. The following things are test:
+
+    1. Default values are set correctly.
+    2. set_config works properly
+    3. Execution order does not change the result e.g. parent is changed
+        if child is executed first.
+
+    Following kind of settings are tested:
+
+    1. Tracked options which are specified via the context, e.g.
+        n_pmts_tpc.
+    2. Untracked options which are specified via the context, e.g.
+        channel_map.
+    3. Settings only specified in the parent and therefore shared by the
+        child.
+    4. Settings only specified for the children but not for the parent.
+    5. Settings which are used in both plugins, but for which the child
+        uses a different value than the parent (child_options)
+    """
+
+    # Special UserWarning which will be raised, but can be "ignored" since it is just a hint for the user
+    warn_message = ('You specified plugin ChildPlugin as a child plugin. Found the option '
+                    '2nd_child_exclusive_option_child with the ending _child which was '
+                    'not specified as a child option. Was this intended?')
+    with pytest.warns(UserWarning, match=warn_message):
+        mystrax = strax.Context(storage=[],
+                                config={'context_option': 4,
+                                        'more_special_context_option': immutabledict(tpc=(0, 4))
+                                        },
+                                register=[Records, Peaks, ParentPlugin, ChildPlugin],
+                                allow_multiprocess=True)
+
+        parent = mystrax.get_array(run_id=run_id, targets='peaks_parent')
+        child = mystrax.get_array(run_id=run_id, targets='peaks_child')
+
+        # ---------------------------------
+        # Checking if default setting of
+        # parent and child are correct:
+        # ---------------------------------
+        # Unchanged values and common values::
+        assert np.all(parent['time'] == child['time']), 'Time field is different for parent and child'
+        assert np.all(parent['length'] == child['length']), 'Length field is different for parent and child'
+        assert np.all(parent['dt'] == child['dt']), 'dt field is different for parent and child'
+
+        m = np.all(parent['max_gap'] == child['max_gap'])
+        assert m, 'parent_unique_option is not equal in parent and child'
+
+        # Values which should have changed:
+        assert np.all(parent['area'] == 2), 'by_child_overwrite_option has changed the value of the parent'
```

Comment:
You really went all the way for these tests! I haven't looked at them in great detail but it looks like you put a lot of effort in it!

Perhaps an idea for the future is to set these numbers above in the script (as a dict?) as if one ever changes a number because we ran into a new testcase we don't have to remember (or ctrl+f the number '2' in the tests to look for) all the places were we have to change the number. Usually you test many numbers (e.g. what if its 0 instead of 2) but I don't feel this is an essential perquisite here.

It seems quite unlikely though that one will change these numbers so I think this will do!

Reply:
Agreed, it has become a bit messy. I will try to check whether I can make it a bit cleaner.

Reply:
Added your suggestion of a default config dict